src/ts/spatialnavigation/seekbarhandler.ts

```diff
@@ -0,0 +1,154 @@
+    const playbackPositionMarker = getPlaybackPositionMarker(seekBarWrapper);
+    const rect = playbackPositionMarker.getBoundingClientRect();
+    const startX = rect.x + (rect.width / 2);
+    const startY = rect.y;
+
+    this.cursorPosition.x = startX;
+    this.cursorPosition.y = startY;
+  }
+
+  private initializeOrUpdateCursorPosition(seekBarWrapper: HTMLElement, direction: Direction): void {
+    if (this.isScrubbing) {
+      this.updateCursorPosition(direction, seekBarWrapper);
+    } else {
+      this.initializeCursorPosition(seekBarWrapper);
+    }
+
+    this.isScrubbing = true;
+  }
+
+  private getCursorPositionMouseEventInit(): MouseEventInit {
+    return {
+      clientX: this.cursorPosition.x,
+      clientY: this.cursorPosition.y,
+    };
+  }
+
+  private dispatchMouseMoveEvent(seekBar: Element): void {
+    seekBar.dispatchEvent(new MouseEvent('mousemove', this.getCursorPositionMouseEventInit()));
+  }
+
+  private readonly onNavigation = (direction: Direction, target: HTMLElement): boolean => {
+    if (!isSeekBarWrapper(target)) {
+      return false;
+    }
+
+    if (direction === Direction.UP || direction === Direction.DOWN) {
+      this.stopSeeking(getSeekBar(target));
+
+      return false;
+    }
+
+    this.initializeOrUpdateCursorPosition(target, direction);
+    this.dispatchMouseMoveEvent(getSeekBar(target));
+
+    return true;
+  };
+
+  private dispatchMouseClickEvent(seekBar: Element): void {
+    const mouseDownHandler = () => {
+      const mouseEventInit = this.getCursorPositionMouseEventInit();
+
+      document.dispatchEvent(new MouseEvent('mouseup', mouseEventInit));
+      this.eventSubscriber.off(seekBar, 'mousedown', mouseDownHandler);
+      this.stopSeeking(seekBar);
+    };
+
+    this.eventSubscriber.on(seekBar, 'mousedown', mouseDownHandler);
+    seekBar.dispatchEvent(new MouseEvent('mousedown'));
+  }
+
+  private stopSeeking(seekBar: Element): void {
+    this.resetCursorPosition();
+    this.isScrubbing = false;
+    this.dispatchMouseLeaveEvent(seekBar);
+  }
+
+  private dispatchMouseLeaveEvent(seekBar: Element): void {
+    seekBar.dispatchEvent(new MouseEvent('mouseleave'));
+  }
+
+  private readonly onAction = (action: Action, target: HTMLElement): boolean => {
+    if (!isSeekBarWrapper(target)) {
+      return false;
+    }
+
+    const seekBar = getSeekBar(target);
+
+    if (action === Action.SELECT && this.isScrubbing) {
+      this.dispatchMouseClickEvent(seekBar);
+      return true;
+    } else if (action === Action.BACK) {
+      this.stopSeeking(seekBar);
+      return true;
+    }
+
+    return false;
+  };
+
+  public release(): void {
+    this.eventSubscriber.release();
+    this.rootNavigationGroup.onAction = undefined;
+    this.rootNavigationGroup.onNavigation = undefined;
+  }
+}
+
+function isSeekBarWrapper(element: HTMLElement): boolean {
+  return Array.from(element.classList).findIndex(className => className.includes('-ui-seekbar')) > -1;
```

Comment:
this would also match `bmpui-ui-seekbar-label`. Isn't that a problem?

Reply:
Fair point. Using a more strict check in 06f9e8eb.

Reply:
Using regex now - addressed in [06f9e8e](https://github.com/bitmovin/bitmovin-player-ui/pull/468/commits/06f9e8eb512a641311758adea1ce73111097ea65)